ci/scripts/detect-docker-updates.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+latest_version=$(curl -I -s https://github.com/docker/docker-ce/releases/latest | grep "location:" | awk '{n=split($0, parts, "/"); print substr(parts[n],2);}' | awk '{$1=$1;print}' | tr -d '\r' | tr -d '\n' )
+
+if [[ $latest_version =~ (beta|rc) ]]; then
+	echo "Skip pre-release versions"
+	exit 0;
+fi
+
+latest="https://download.docker.com/linux/static/stable/x86_64/docker-$latest_version.tgz"
+current=$( git-repo/ci/images/get-docker-url.sh )
+
+if [[ $current = $latest ]]; then
+	echo "Already up-to-date"
+	exit 0;
+fi
+
+
+issue_title="Upgrade to Docker $latest_version in CI"
+
+existing_tasks=$( curl -s https://api.github.com/repos/${GITHUB_ORGANIZATION}/${GITHUB_REPO}/pulls\?labels\=type:%20task\&state\=open\&creator\=spring-buildmaster )
+existing_upgrade_issues=$( echo "$existing_tasks" | jq -c --arg TITLE "$issue_title" '.[] | select(.title==$TITLE)' )
+
+if [[ ${existing_upgrade_issues} = "" ]]; then
+	# Create changes in dedicated branch
+	branch="ci-docker-$latest_version"
+	git config user.name "Spring Buildmaster" > /dev/null
+	git config user.email "buildmaster@springframework.org" > /dev/null
+	git checkout -b "$branch" origin/master > /dev/null
+	sed -i "s/version=.*/version=\"$latest_version\"/" git-repo/ci/images/get-docker-url.sh
+	git add git-repo/ci/images/get-docker-url.sh > /dev/null
+	git commit -m "$issue_title" > /dev/null
+	git push origin "$branch" > /dev/null
```

Comment:
Do we maybe need to switch to the master branch again afterwards?

Reply:
I don't think we can push from here as it will ask for credentials. We should probably commit inside the task and push using the `git-resource` in `pipeline.yml`. If there was no commit, it should be a no-op. Once the git-resource is used to push, we can trigger another task that creates the pull-request. This task config can then be shared between all the automatic detection jobs that we have (jdk, ubuntu and docker). It can take the branch name and issue title as params. What do you think @dreis2211?

Reply:
I'm not quite sure if I understand what you want to share exactly? Everything inside this if (so line 24-46) or just the curl command in line 36-43?
I must admit the Concourse setup you're asking for is not completely clear to me (although I generally understand the plan), but my Concourse experience is limited. I'd probably need a little bit of help with that, I guess.